packages/dds/tree/src/core/tree/detachedFieldIndexCodec.ts

```diff
@@ -42,9 +50,13 @@ export function makeDetachedNodeToFieldCodec(
 				"Cannot decode detached field index without revision tag codec",
 			);
 			const map = new Map();
-			for (const [rootRanges, revision] of parsed.data) {
-				const innerMap = new Map(rootRanges);
-				map.set(revision, innerMap);
+			for (const rootsForRevision of parsed.data) {
+				const innerMap = new Map<number, ForestRootId>(
+					rootsForRevision.length === 2
```

Comment:
Might be worth asserting the length === 3 in the other branch of the ternery. I can't tell how type-safe this is, though.

Reply:
The codec already validates that, so it will throw if that isn't the case.